staging/src/k8s.io/apimachinery/pkg/runtime/codec.go

```diff
@@ -45,14 +46,23 @@ func NewCodec(e Encoder, d Decoder) Codec {
 
 // Encode is a convenience wrapper for encoding to a []byte from an Encoder
 func Encode(e Encoder, obj Object) ([]byte, error) {
-	// TODO: reuse buffer
-	buf := &bytes.Buffer{}
+	buf := bufPool.Get().(*bytes.Buffer)
+	defer bufPool.Put(buf)
+
+	buf.Reset()
```

Comment:
Reset() puts the starting point back to 0, but retains the previously allocated memory to write into:

> Reset resets the buffer to be empty, but it retains the underlying storage for use by future writes. Reset is the same as Truncate(0).

I don't think this makes it safe to write to a buffer we previously returned a byte slice from

Reply:
ok , just remove the todo